lib/http/timeout/null.rb

```diff
@@ -56,19 +56,31 @@ def write(data)
       private
 
       # Retry reading
-      def rescue_readable(timeout = read_timeout)
+      def rescue_readable(klass, timeout = read_timeout)
         yield
       rescue IO::WaitReadable
         retry if @socket.to_io.wait_readable(timeout)
-        raise TimeoutError, "Read timed out after #{timeout} seconds"
+        raise klass, "#{error_name_for_class(klass)} timed out after #{timeout} seconds"
       end
 
       # Retry writing
-      def rescue_writable(timeout = write_timeout)
+      def rescue_writable(klass, timeout = write_timeout)
         yield
       rescue IO::WaitWritable
         retry if @socket.to_io.wait_writable(timeout)
-        raise TimeoutError, "Write timed out after #{timeout} seconds"
+        raise klass, "#{error_name_for_class(klass)} timed out after #{timeout} seconds"
```

Comment:
CodeClimate [does not like](https://codeclimate.com/github/httprb/http/pull/471) the similarity of these methods. I'm not sure abstracting out would improve the overall quality though, thoughts?

Reply:
I think it's fine for not to ignore this code climate regression. We can think about that later. I don't see that abstracting this will help atm as well.